### PR TITLE
test(portal): rewrite all .old tests

### LINKED ISSUE #11056: Re-implement broken portal tests
Due to #10505, many portal domain tests are now broken. This issue tracks work to resolve those.

### PR DESCRIPTION
**Pull request by jamilbk**
Restores/rewrites all of the `.old` tests from before the major IdP refactor.

Some application code was changed in order to fix failing assertions (read: bugs).

--- 
Fixes #11056

### CODE REVIEW

**Comment by jamilbk** [comment]
Draft because stacked on #12431

File: elixir/lib/portal_web/live/clients/edit.ex [hunk 1]
```diff
@@ -60,6 +61,7 @@ defmodule PortalWeb.Clients.Edit do
   def handle_event("change", %{"client" => attrs}, socket) do
     changeset =
       update_changeset(socket.assigns.client, attrs)
+      |> Portal.Client.changeset()
       |> Map.put(:action, :insert)
```
[the remark this answers is not in the document]
Comment (jamilbk) [reply elixir/lib/portal_web/live/clients/edit.ex:65]: This is not correct - Portal.Safe runs changesets on modules if they're defined, before proceeding.

File: elixir/lib/portal/repo/paginator.ex [hunk 2]
```diff
@@ -326,6 +327,17 @@ defmodule Portal.Repo.Paginator do
...
 
+  defp validate_cursor_values(cursor_fields, values) do
+    valid? =
+      Enum.zip(cursor_fields, values)
+      |> Enum.all?(fn
+        {{_, :asc, _}, nil} -> false
+        _ -> true
+      end)
+
+    if valid?, do: :ok, else: {:error, :invalid_cursor}
```
[the remark this answers is not in the document]
Comment (jamilbk) [reply elixir/lib/portal/repo/paginator.ex:338]: This is changing soon with #5732 so not worth fixing edge cases here.

File: elixir/lib/portal/presence.ex
Comment (jamilbk) [inline elixir/lib/portal/presence.ex:1]: This is mostly a change to make the presence tests less flaky.

File: elixir/lib/portal_web/live/clients/edit.ex
Comment (jamilbk) [inline elixir/lib/portal_web/live/clients/edit.ex:1]: Some minor refactoring in this module.

File: elixir/lib/portal/repo/paginator.ex
Comment (jamilbk) [inline elixir/lib/portal/repo/paginator.ex:1]: Prevents an invalid cursor from raising `ArgumentError`

File: elixir/lib/portal_web/live/sites/new.ex
Comment (jamilbk) [inline elixir/lib/portal_web/live/sites/new.ex:1]: Minor cleanup

File: elixir/lib/portal_web/live/sites/index.ex [hunk 3]
```diff
@@ -203,7 +209,7 @@ defmodule PortalWeb.Sites.Index do
     socket =
       socket
       |> assign(internet_resource: internet_resource)
-      |> assign(internet_site: internet_resource.site)
+      |> assign(internet_site: internet_resource && internet_resource.site)
```
Comment (jamilbk) [inline elixir/lib/portal_web/live/sites/index.ex]: Prevents the user from seeing a 500 if for some reason there is no internet resource for this account.

Comment (bmanifold) [reply elixir/lib/portal_web/live/sites/index.ex]: If there's not an Internet Resource, isn't that a problem?  Is this a good usage of "let it crash" so we know we need to go fix something?

Comment (jamilbk) [reply elixir/lib/portal_web/live/sites/index.ex]: We can both get notified and not disrupt the LiveView. That said I don't think this is possible to happen in practice because this is covered by multiple tests.

I'll just revert.

Comment (jamilbk) [reply elixir/lib/portal_web/live/sites/index.ex]: Fixed

**Review by bmanifold — COMMENTED** [review]
In general it looks good, but it looks like there are a few failing tests.

**Comment by jamilbk** [comment]
Tests have been fixed

**Review by bmanifold — APPROVED** [review]
